dipy/workflows/io.py

```diff
@@ -112,6 +118,44 @@ def run(self, input_files,
                 logging.info('Total number of non-unit bvectors {0}\n'
                              .format(ncl1))
 
+            if extension in ['.trk', '.tck', '.trx', '.vtk', '.vtp', '.fib',
+                             '.dpy']:
+
+                sft = None
+                if extension in ['.trk', '.trx']:
+                    sft = load_tractogram(input_path, 'same',
+                                          bbox_valid_check=False)
+                else:
+                    sft = load_tractogram(input_path, reference,
+                                          bbox_valid_check=False)
+
+                lengths = [len(s) for s in sft.streamlines]
+                lengths_mm = list(length(sft.streamlines))
+
+                sft.to_voxmm()
+                steps = [np.sqrt(np.sum(np.diff(s, axis=0) ** 2, axis=1))
```

Comment:
Maybe get the lengths and steps in one loop through instead of the two list comprehensions?

Reply:
Thanks! I updated the code.